### PR TITLE
fix shellcheck warnings and add `make lint` to invoke it

### LINKED ISSUE #34: shellcheck
https://github.com/koalaman/shellcheck

### PR DESCRIPTION
**Pull request by shamrin**
* invoke `shellcheck` via `make lint`
* fix all the warnings

Fixes #34.

### CODE REVIEW

File: shmig [hunk 1]
```diff
@@ -28,26 +29,19 @@ COLOR="never"
 # we assume color output is not used
 
 RED=""
-GREEN=""
-YELLOW=""
-BLUE=""
-MAGENTA=""
```
Comment (shamrin) [inline shmig]: Fixes https://www.shellcheck.net/wiki/SC2034

File: shmig [hunk 2]
```diff
@@ -28,26 +29,19 @@ COLOR="never"
...
 LMAGENTA=""
 LCYAN=""
-LWHITE=""
 CLEAR=""
 BOLD=""
 
 # some helpers
 warn(){
-  echo -e "$ME: $@" >&2
+  echo -e "$ME: $*" >&2
```
Comment (shamrin) [inline shmig]: Fixes https://www.shellcheck.net/wiki/SC2145

File: shmig [hunk 3]
```diff
@@ -166,8 +160,7 @@ __generic_checker(){
 
   TMPF=/tmp/$(((RANDOM<<15)|RANDOM)).$$.out
   rm -f $TMPF
-  "${TYPE}_cli" "$show" > $TMPF
-  if [ $? -ne 0 ]
+  if ! "${TYPE}_cli" "$show" > $TMPF
```
Comment (shamrin) [inline shmig]: Fixes https://www.shellcheck.net/wiki/SC2181

File: shmig [hunk 4]
```diff
@@ -501,13 +495,14 @@ pending_migrations(){
 previous_migrations(){
   local version=""
 
-  while read version
+  while read -r version
```
Comment (shamrin) [inline shmig]: https://www.shellcheck.net/wiki/SC2162

File: shmig [hunk 5]
```diff
@@ -644,14 +640,15 @@ do
     ?)   exit 1                    ;;
   esac
 done
-shift $(($OPTIND - 1))
+shift $((OPTIND - 1))
```
Comment (shamrin) [inline shmig]: Fixes https://www.shellcheck.net/wiki/SC2004

File: shmig [hunk 6]
```diff
@@ -644,14 +640,15 @@ do
...
 
 # if config exists
 if [[ -e $CONFIG ]]
 then
   # any error in configuration file should cause failure
   # to avoid potential database corruption
   trap 'die "Configuration failed"' ERR
+  # shellcheck source=/dev/null
```
Comment (shamrin) [inline shmig]: Fixes https://www.shellcheck.net/wiki/SC1090

File: shmig [hunk 7]
```diff
@@ -668,13 +665,14 @@ if [ -e "$LOCAL_CONFIG" ]; then
   # any error in configuration file should cause failure
   # to avoid potential database corruption
   trap 'die "Local configuration failed"' ERR
+  # shellcheck source=/dev/null
   . "$LOCAL_CONFIG"
```
Comment (shamrin) [inline shmig]: Fixes https://www.shellcheck.net/wiki/SC1090

File: shmig [hunk 8]
```diff
@@ -212,18 +205,18 @@ __generic_status(){
 
 ##### MySQL
 mysql_cli(){
-  local args="-N -A -B $ARGS $DATABASE"
+  local args=(-N -A -B "$ARGS" "$DATABASE")
```
Comment (shamrin) [inline shmig]: Changed to array in order to fix https://www.shellcheck.net/wiki/SC2086

**Comment by mbucc** [comment]
Alexy, thanks so much for this work!

**Pull request merged**